### ISSUE TITLE
Error while compiling source

### ISSUE THREAD

**Issue by klangner** [tantivy-search/tantivy-cli#39]
I have fresh version cloned from this repo and when I try to run **cargo build** I get the following error:
```
error[E0034]: multiple applicable items in scope
   --> /Users/klangner/.cargo/registry/src/github.com-1ecc6299db9ec823/downcast-0.9.2/src/lib.rs:120:38
    |
120 |     fn is_type(&self) -> bool { self.type_id() == TypeId::of::<T>() }
    |                                      ^^^^^^^ multiple `type_id` found
    |
note: candidate #1 is defined in the trait `Any`
   --> /Users/klangner/.cargo/registry/src/github.com-1ecc6299db9ec823/downcast-0.9.2/src/lib.rs:29:5
    |
29  |     fn type_id(&self) -> TypeId { TypeId::of::<Self>() }
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: to disambiguate the method call, write `Any::type_id(&self)` instead
note: candidate #2 is defined in the trait `std::any::Any`
    = help: to disambiguate the method call, write `std::any::Any::type_id(&self)` instead

error: aborting due to previous error

For more information about this error, try `rustc --explain E0034`.
```

I have MacOs and rust version 1.34

I can also compile without problems **tantivy**.

It looks that the problem could be fixed with newer version of dependencies. I can fix it and create PR. This will require upgrading **tantivy** and do some changes since the API changed.

**Comment by fulmicoton** [comment]
I have been ablle to reproduce. 

tantivy-cli has simply not been updated for a long time and does not compile with recent compilers. Fixing that today.

**Comment by fulmicoton** [comment]
Fixed in master, and publishing tantivy-cli@0.9 in a minute.
